src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.Azure.Common;
+using Azure;
+using Azure.Core;
+
+namespace Aspire.Azure.AI.OpenAI;
+
+/// <summary>
+/// The settings relevant to accessing Azure OpenAI or OpenAI.
+/// </summary>
+public sealed class AzureOpenAISettings : IConnectionStringSettings
+{
+    private const string ConnectionStringEndpoint = "Endpoint";
+    private const string ConnectionStringKey = "Key";
+
+    /// <summary>
+    /// Gets or sets a <see cref="Uri"/> referencing the Azure OpenAI endpoint.
+    /// This is likely to be similar to "https://{account_name}.openai.azure.com".
+    /// </summary>
+    /// <remarks>
+    /// Leave empty and provide a <see cref="Key"/> value to use a non-Azure OpenAI inference endpoint.
+    /// Used along with <see cref="Credential"/> or <see cref="Key"/> to establish the connection.
+    /// </remarks>
+    public Uri? Endpoint { get; set; }
+
+    /// <summary>
+    /// Gets or sets the credential used to authenticate to the Azure OpenAI resource.
+    /// </summary>
+    public TokenCredential? Credential { get; set; }
+
+    /// <summary>
+    /// Gets or sets the key to use to authenticate to the Azure OpenAI endpoint.
+    /// </summary>
+    /// <remarks>
+    /// When defined it will use an <see cref="AzureKeyCredential"/> instance instead of <see cref="Credential"/>.
+    /// </remarks>
+    public string? Key { get; set; }
+
+    /// <summary>
+    /// Gets or sets a boolean value that indicates whether the OpenTelemetry tracing is enabled or not.
+    /// </summary>
+    /// <value>
+    /// The default value is <see langword="true"/>.
+    /// </value>
+    public bool Tracing { get; set; } = true;
+
+    void IConnectionStringSettings.ParseConnectionString(string? connectionString)
+    {
+        var connectionBuilder = new DbConnectionStringBuilder
+        {
+            ConnectionString = connectionString
+        };
+
+        if (connectionBuilder.ContainsKey(ConnectionStringEndpoint) && Uri.TryCreate(connectionBuilder[ConnectionStringEndpoint].ToString(), UriKind.Absolute, out var serviceUri))
```

Comment:
I'm concerned that we are introducing our own connection string format for Azure OpenAI. I think this would be better if it was defined by the Azure SDK.

@tg-msft - Is there any possibility to get Azure.AI.OpenAI to support a connection string format? .NET Aspire really goes "all in" on connection strings - it is how components get connection information like this passed between the AppHost and the individual service projects.

cc @davidfowl @mitchdenny @DamianEdwards

Reply:
I think this is going to be a somewhat common theme. Where a clear connection string format exists for a particular tech we should use that.

We have issues around this for Cosmos as well where there is a connection string format for AccountKey based auth, but nothing established for MSI auth.

Reply:
Would it not be confusing for people coming to it that are familiar with AOAI anyway? AOAI doesn't have a "connection string", it has an endpoint and key, so this is inventing something that doesn't exist, resulting in someone having to learn the "Aspire AOAI way" rather than being familiar with how you natively connect to AOAI.

Reply:
@sebastienros and I came up with these 3 options of how to pass the connection information into the component:

```json
{
  "ConnectionStrings": {
    // option 1.a
    "myOpenAI": "Endpoint=https://url;Key=123",

    // option 1.b
    "myOpenAI_Endpoint": "https",
    "myOpenAI_Key": "123"
  },
  // option 2
  "ConnectionProperties": {
    "myOpenAI": {
      "Endpoint": "https",
      "Key": "123"
    }
  },
  "Aspire": {
    "Azure": {
      "AI": {
        "OpenAI": {
          // option 3
          "myOpenAI": {
            "Endpoint": "https://url",
            "Key": "123"
          },
          "Tracing": false
        }
      }
    }
  }
}
```

Options 1.a and 1.b use the ConnectionStrings section.
Option 2 introduces a brand new "ConnectionProperties" section, which is like ConnectionSTrings, but can be any format, not just strings.
Option 3 puts the connection information in the Azure OpenAI specific section, always under the "connectionName" (in this case "myOpenAI").

IMO - the only viable options are `option 1.a` and `option 3`. I don't think we should do `1.b` or `2`.

Reply:
I like 1.a

Reply:
I like 1.a (nb: it's what is currently implemented in the PR)

Reply:
Coming at this from the perspective of "I use Azure OpenAI and now want to explore Aspire" my vote would be for option 2 or 3 (although 1.b would also be viable).

The reason is that there is no "connection string" construct in Azure OpenAI, meaning that I'm not going to find the value for 1.a anywhere, instead I'm having to make something up that's Aspire specific.

Reply:
> Is there any possibility to get Azure.AI.OpenAI to support a connection string format?

The Azure SDK Arch Board has rejected connection strings not present in the Portal whenever it's come up in the past.  See https://github.com/Azure/azure-sdk/issues/535, https://github.com/Azure/azure-sdk/issues/1701, and https://github.com/Azure/azure-sdk/issues/3288 for examples.  It's [pretty much just Storage, Messaging, and Communication services](https://grep.app/search?q=public%20%5BA-Za-z0-9%5D%2B%5C%28string%20connectionString%5C%2C%20%5BA-Za-z0-9.%5D%2BClientOptions&regexp=true&filter[repo][0]=Azure/azure-sdk-for-net) in [Track 2](https://aka.ms/azsdk/intro) that support connection strings today.

Even then, if we decided to add this in `Azure.AI.OpenAI` we would want to do so uniformly across languages.  .NET/Java/JS all have their own Azure specific package that works with AOAI (and with OAI for the intersection of shared features).  Python has an inverted approach where they added AOAI support to the official OAI package that we don't own.

I was leaning toward Options 2 and 3 when thinking about IConfig support in Azure clients.  Connection strings not supported in the Portal feel like a worse experience for customers (i.e., I'll get intellisense on the JSON in Options 2 and 3 but I'm assuming I'll be left reading docs to construct a synthetic connection string?).

Reply:
Option 2 was the first option we came up with but we were afraid of being too non-standard, so it seems like the most agreed upon choice now. Unless some higher power disagrees I will move forward with this option, but I will still merge it as it is (connection string) so it doesn't block the Open AI component (we won't ship this connection string).